lib/string_decoder.js

```diff
@@ -46,9 +50,12 @@ function StringDecoder(encoding) {
   this.lastChar = Buffer.allocUnsafe(nb);
 }
 
+// TODO(addaleax): This method should not accept strings as input.
```

Comment:
I don't understand the comment here. Is the suggestion that in the future it should throw on a string? Otherwise the comment seems at odds with the string check below.